src/pages/index.js

```diff
@@ -0,0 +1,361 @@
+    localStorage.setItem("likedCards", JSON.stringify({})); // Reset storage
+  }
+
+  // Apply liked state
+  if (likedCards[data._id]) {
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  //add the event listener
+  cardLikedBtn.addEventListener("click", (evt) => handleLikeCard(evt, data));
+  deleteCardBtn.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  //add event listener for the card image
+  cardImage.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaption.textContent = data.name;
+  });
+  // console.log("Generated Card:", cardElement); // Debugging
+  return cardElement;
+}
+
+function handleLikeCard(evt, data) {
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+
+  api
+    .handleLike(data._id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-button_liked");
+
+      let likedCards = JSON.parse(localStorage.getItem("likedCards")) || {};
+
+      if (isLiked) {
+        delete likedCards[data._id];
+      } else {
+        likedCards[data._id] = true;
+      }
+      localStorage.setItem("likedCards", JSON.stringify(likedCards));
+    })
+    .catch((err) => {
+      console.error("Error updating like:", err);
+    });
+}
+
+previewCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  // Add event listener to close modal on Escape key
+  document.addEventListener("keydown", closeOnEscape);
+  modal.addEventListener("mousedown", closeOnOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  // Remove event listener when modal is closed
+  document.removeEventListener("keydown", closeOnEscape);
+  modal.removeEventListener("mousedown", closeOnOverlay);
+}
+
+function closeOnEscape(evt) {
+  if (evt.key === "Escape") {
+    const openModal = document.querySelector(".modal_opened");
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function closeOnOverlay(evt) {
+  if (evt.target.classList.contains("modal_opened")) {
+    closeModal(evt.target);
+  }
+}
+
+// function handleCardSubmit(evt) {
+//   evt.preventDefault();
+
+//   const inputValues = {
+//     name: cardNameInput.value,
+//     link: cardLinkInput.value,
+//   };
+//   const cardElement = getCardElement(inputValues);
+//   cardsList.prepend(cardElement);
+
+//   cardForm.reset();
+//   disableBtn(cardSubmitBtn, validationConfig);
+//   closeModal(cardModal);
+// }
+
+// ToDo: add the api call to create a new card
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Saving...", "Save");
+  api
+
+    .editAvatar({
+      avatar: avatarForm.querySelector("#profile-avatar-input").value,
```

Comment:
`avatarForm.querySelector("#profile-avatar-input")`


Please, find the input **outside** the function only once as `avatarInput`, and use it here


```js
      avatar: avatarInput.value,

```

Reply:
* Another remark hasn't been corrected https://github.com/wedderburn95/se_project_spots/pull/1#discussion_r2024222605